hardware_interface/test/interface_manager_test.cpp

```diff
@@ -84,6 +84,93 @@ TEST(InterfaceManagerTest, InterfaceRewriting)
   EXPECT_EQ(2, foo_iface_ptr->foo);
 }
 
+struct TestCombinationHandle {
+  TestCombinationHandle(const std::string& name) : name{name} {}
+
+  std::string getName() const
+  {
+    return name;
+  }
+
+  std::string name;
+};
+
+struct TestCombinationInterface : public ResourceManager<TestCombinationHandle>
+{
+
+};
+
+TEST(InterfaceManagerTest, InterfaceCombination)
+{
+  // Create two InterfaceManagers, each containing a simple interface and a ResourceManager interface,
+  // and an additional simple interface for the second one
+  InterfaceManager root_mgr;
+  InterfaceManager leaf_mgr;
+
+  FooInterface foo_root_iface(1);
+  root_mgr.registerInterface(&foo_root_iface);
+  FooInterface foo_leaf_iface(2);
+  leaf_mgr.registerInterface(&foo_leaf_iface);
+
+  BazInterface baz_leaf_iface;
+  baz_leaf_iface.baz = 4.2;
+  leaf_mgr.registerInterface(&baz_leaf_iface);
+
+  TestCombinationInterface combi_root_iface;
+  TestCombinationHandle combi_root_handle("foo_handle");
+  combi_root_iface.registerHandle(combi_root_handle);
+  root_mgr.registerInterface(&combi_root_iface);
+
+  TestCombinationInterface combi_leaf_iface;
+  TestCombinationHandle combi_leaf_handle("bar_handle");
+  combi_leaf_iface.registerHandle(combi_leaf_handle);
+  leaf_mgr.registerInterface(&combi_leaf_iface);
+
+  // Now register leaf_mgr to root_mgr
+  root_mgr.registerInterfaceManager(&leaf_mgr);
+
+  // Querying FooInterface should not work anymore, as this would require combining the interfaces
+  FooInterface* foo_iface_ptr = root_mgr.get<FooInterface>();
+  EXPECT_EQ(nullptr, foo_iface_ptr);
+
+  // BazInterface should still work, as there is only one interface of this type
+  BazInterface* baz_iface_ptr = root_mgr.get<BazInterface>();
+  EXPECT_NE(nullptr, baz_iface_ptr);
+  EXPECT_EQ(4.2, baz_iface_ptr->baz);
+
+  // Check presence of handles in combined interface
+  TestCombinationInterface* combi_iface_ptr = root_mgr.get<TestCombinationInterface>();
+  std::vector<std::string> combi_handle_names = combi_iface_ptr->getNames();
+  EXPECT_NE(nullptr, combi_iface_ptr);
```

Comment:
Check `combi_iface_ptr` before dereferencing it, otherwise the test could crash in a way that doesn't give us a useful failure message

Reply:
Nice catch, applied in rebased commits